File: 1208. Get Equal Substrings Within Budget/equalSubstring.py
class Solution:
  def equalSubstring(self, s: str, t: str, maxCost: int) -> int:
    currentCost = 0
    left = 0
    res = 0

    for right in range(len(s)):
      currentCost += abs(ord(s[right]) - ord(t[right]))

      while currentCost > maxCost:
        currentCost -= abs(ord(s[left]) - ord(t[left]))
        left += 1

      res = max(res, right - left + 1)

    return res
  

s = "abcd"
t = "bcdf"
maxCost = 3

solution = Solution()
print('equalSubstring: ', solution.equalSubstring(s, t, maxCost))

File: 1208. Get Equal Substrings Within Budget/equalSubstring.js
/**
 * @param {string} s
 * @param {string} t
 * @param {number} maxCost
 * @return {number}
 */
const equalSubstring = (s, t, maxCost) => {
  let currentCost = 0;
  let left = 0;
  let res = 0;

  for (let right = 0; right < s.length; right++) {
    currentCost += Math.abs(s[right].charCodeAt(0) - t[right].charCodeAt(0));

    while (currentCost > maxCost) {
      currentCost -= Math.abs(s[left].charCodeAt(0) - t[left].charCodeAt(0));
      left++;
    }
    
    res = Math.max(res, right - left + 1);
  }

  return res;
};


const s = "abcd";
const t = "bcdf";
const maxCost = 3;

console.log('equalSubstring: ', equalSubstring(s, t, maxCost));
